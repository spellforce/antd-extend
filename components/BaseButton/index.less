.base-btn {
  border-radius: 4px;
}
