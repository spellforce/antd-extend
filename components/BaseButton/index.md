```js
const handleClick = () => {
  console.log('clicked');
};
<>
  <BaseButton onClick={handleClick}>主按钮</BaseButton>
  <pre />
  <BaseButton type="ghost">次按钮</BaseButton>
  <pre />
  <BaseButton type="dashed">虚线按钮</BaseButton>
  <pre />
  <BaseButton type="link">链接按钮</BaseButton>
  <pre />
  <BaseButton type="text">文本按钮</BaseButton>
</>;
```
