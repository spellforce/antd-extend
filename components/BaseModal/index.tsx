import React, { useState } from 'react';
import { Button } from 'antd';
import './index.less';

interface Props {
  /**
   * 按钮类型 默认primary
   */
  type?: 'primary' | 'ghost' | 'dashed' | 'link' | 'text' | 'default';
  /**
   * 点击事件  有交互的时候用这个
   */
  onClick?: Function;
}

/**
 * Button 组件
 * @link [antd button](https://ant.design/components/button-cn/)
 */

function BaseButton(props: Props) {
  const [loading, setLoading] = useState(false);
  const { onClick, type } = props;
  const handleClick = async () => {
    if (!onClick) return;
    if (loading) return;
    setLoading(true);
    try {
      await onClick();
    } finally {
      setLoading(false);
    }
  };

  return (
    <Button {...props} type={type} loading={loading} onClick={handleClick} className="base-btn" />
  );
}

BaseButton.defaultProps = {
  type: 'primary',
};

export default BaseButton;
