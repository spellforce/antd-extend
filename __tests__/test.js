import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import BaseButton from '../components/BaseButton';
import BaseModal from '../components/BaseModal';

Enzyme.configure({ adapter: new Adapter() });
describe('BaseButton', () => {
  const handleClick = jest.fn();
  const buttonwrapper = shallow(<BaseButton onClick={handleClick}>hello</BaseButton>);
  const modalwrapper = Enzyme.shallow(
    <BaseModal>
      <p>Some contents...</p>
    </BaseModal>,
  );
  it('should have rendered BaseButton', () => {
    expect(buttonwrapper).toMatchSnapshot();
  });
  it('should have rendered BaseModal', () => {
    expect(modalwrapper).toMatchSnapshot();
  });
  it('should excute click event', () => {
    buttonwrapper.find('.base-btn').simulate('click');
    expect(handleClick).toBeCalled();
  });
});
