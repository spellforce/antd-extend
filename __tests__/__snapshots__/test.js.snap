// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton should have rendered BaseButton 1`] = `ShallowWrapper {}`;

exports[`BaseButton should have rendered BaseModal 1`] = `ShallowWrapper {}`;
